### DISCUSSION TITLE
What's the easiest way to collaborate async with a nontechnical coauthor?

### DISCUSSION THREAD

**Discussion by quesada** [zim-desktop-wiki/zim-desktop-wiki#1836] [Q&A]
I'm not looking for 'google docs like' collaboration.
Just Reading and adding comments could be enough.
since these are plain text files, is there any online service that would give a web interface to edit text files?
Maybe someone here has found a workaround. Like having a public link with dropbox or similar?

**Comment by introt** [comment]
> is there any online service that would give a web interface to edit text files?

GitHub ;)

No, but seriously - it's literally built for async collaboration on text files. You don't even need to know how to do pull requests, regular commits can be commented on on a line-by-line basis:

![image](https://user-images.githubusercontent.com/26069787/147489284-fee5dde0-616b-4bf0-8fd6-c9d27cdaafa0.png)

I'm assuming you're the one reading and adding comments - just set up their Zim's Version control plugin with post-commit hooks that push their changes to the (private) repo, and they won't have to do a thing! GitHub's GUI is also pretty user friendly, so two-way collaboration shouldn't be too hard. Or just have them edit the via the online interface - though at that point I must ask why you'd want to use Zim instead of just writing MarkDown files inside GitHub.

If you never need to work on the same files at the same time, a shared "dropbox or similar" folder would also work. For real-time collaboration I'd suggest looking at https://etherpad.org/ etc.

**Reply by introt** [reply]
Non-personal wikis in general are async collaboration on text, but not necessarily text _files_; you might find a non-personal wiki software more suitable for your purposes. GitHub again provides one (see eg. https://github.com/jaap-karssenberg/zim-wiki/wiki for example), but there are many others. For a nontechnical coauthor I'd suggest looking for one with a WYSIWYG editor, as I've seen many first-timers struggle with basic markup syntax.

**Reply by introt** [reply]
See also: https://github.com/zim-desktop-wiki/zim-desktop-wiki/pull/688, "3-way merge functionality"

**Comment by quesada** [comment]
One other way to collaborate (outside bigCo's influence) is https://cryptpad.fr/
I'd love a plugin that would create a https://cryptpad.fr/ doc from a zim doc, invite a collaborator, and save/export the resulting doc into zim (without breaking zimwiki links if any)

**Reply by quesada** [reply]
Yet another way is to use markdown and a collaborative editor. There's hackMD but also https://hedgedoc.org/, which is an open source library that provides exactly that.  Not directly usable but wonderful in case one would want to create a backend that would interface with zim directly (scifi, right now). 
About markdown support: I'm using now @pfalcon fork (https://github.com/pfalcon/zim-desktop-wiki-markdown) with it! it seems to be usable, he uses it for day to day duty. There are some gitches but it fees mature enough that it could be a great candidate to be included into the main repo.
